src/UserStateSerializer.php

```diff
@@ -37,60 +57,70 @@ public function __construct($path)
      */
     public function serialize(UserState $state)
     {
-        return file_put_contents($this->path, json_encode([
-            'completed_exercises'   => $state->getCompletedExercises(),
-            'current_exercise'      => $state->getCurrentExercise(),
-        ]));
+        $saveFile = sprintf('%s/%s', $this->path, static::SAVE_FILE);
+
+        if (!file_exists($saveFile)) {
+            $data = [];
+            $data[$this->workshopName] = [
```

Comment:
this line and the one in the else is the same... maybe just do the initialisation of $data in a short ternary ?